adapters/repos/db/shard_lazyloader.go

```diff
@@ -308,13 +308,6 @@ func (l *LazyLoadShard) drop() error {
 				idx.vectorIndexUserConfig, idx.vectorIndexUserConfigs)
 		}
 
-		// cleanup queue
-		if l.shardOpts.indexCheckpoints != nil {
-			if err := l.shardOpts.indexCheckpoints.Drop(); err != nil {
```

Comment:
resetting the checkpoint for the deleted shard might be needed? (in case the shard is re-created)
`func (c *Checkpoints) Update(shardID, targetVector string, id uint64)` with docID=0 ?

Reply:
Good point! That would mean that this scenario is possible:
- lazyload shard A
- shard A loaded
- index stuff
- restart the node
- delete the shard (before it's loaded)
- ---> we need to clean up the checkpoint here

I'll fix that.

Out of curiosity, do we ever re-create shards with the same ID?

Reply:
I would think it could be possible with multi-tenants, not fully sure if that's currently possible but as far as I understand there shouldn't be a technical reason for not being able to do so. So the checkpoint reset would be to support such a case